lib/ansible/modules/network/radware/vdirect_file.py

```diff
@@ -0,0 +1,243 @@
+    version = pkg_resources.get_distribution("vdirect_client").version
+    if version.startswith("4.1.1"):
+        HAS_REST_CLIENT_VERSION = True
+    else:
+        HAS_REST_CLIENT_VERSION = False
+except ImportError:
+    HAS_REST_CLIENT = False
+
+TEMPLATE_EXTENSION = '.vm'
+WORKFLOW_EXTENSION = '.zip'
+
+
+class VdirectFile(object):
+    def __init__(self, params):
+        self.client = rest_client.RestClient(params['vdirect_ip'],
+                                             params['vdirect_user'],
+                                             params['vdirect_password'],
+                                             wait=params['vdirect_wait'],
+                                             secondary_vdirect_ip=params['vdirect_secondary_ip'],
+                                             https_port=params['vdirect_https_port'],
+                                             http_port=params['vdirect_http_port'],
+                                             timeout=params['vdirect_timeout'],
+                                             https=params['vdirect_https'],
+                                             strict_http_results=params['vdirect_strict_http_results'],
+                                             ssl_verify_context=params['vdirect_ssl_verify_context'])
+
+    def upload(self, fqn):
+        if fqn.endswith(TEMPLATE_EXTENSION):
+            template_name = os.path.basename(fqn)
+            template = rest_client.Template(self.client)
+            runnable_file = open(fqn, 'r')
+            file_content = runnable_file.read()
+
+            result = template.create_from_source(file_content, template_name, fail_if_invalid=True)
+            if result[rest_client.RESP_STATUS] == 409:
+                template.upload_source(file_content, template_name, fail_if_invalid=True)
+                result = "Template updated"
+            else:
+                result = "Template created"
+        elif fqn.endswith(WORKFLOW_EXTENSION):
+            workflow = rest_client.WorkflowTemplate(self.client)
+
+            runnable_file = open(fqn, 'rb')
+            file_content = runnable_file.read()
+            result = workflow.create_template_from_archive(file_content, fail_if_invalid=True)
+            if result[rest_client.RESP_STATUS] == 409:
+                workflow.update_archive(file_content, os.path.splitext(os.path.basename(fqn))[0])
+                result = "Workflow template updated"
+            else:
+                result = "Workflow template created"
+        else:
+            result = "The file_name parameter must have " \
+                     "velocity script (.vm) extension or ZIP archive (.zip) extension"
+        return result
+
+
+def main():
+
+    if not HAS_REST_CLIENT:
+        raise ImportError("The python vdirect-client module is required")
+    elif not HAS_REST_CLIENT_VERSION:
+        raise ImportError("The python vdirect-client module version should be 4.1.1 and above")
+
+    meta_args = dict(
```

Comment:
1) Please put each option on a single line
2) use `type='int'` for ports, timeouts, etc
3) use `type='bool`' for booleans (ssl, vdirect_strict_http_results, etc)
4) Are any of these options required?